modin/pandas/dataframe.py

```diff
@@ -1240,16 +1234,28 @@ def itertuples(self, index=True, name="Pandas"):
         Returns:
             A tuple representing row data. See args for varying tuples.
         """
-        index_iter = iter(self.index)
-
-        def itertuples_builder(df):
-            df.columns = self.columns
-            df.index = [next(index_iter)]
-            return df.itertuples(index=index, name=name)
+        from collections import namedtuple
+
+        if name is not None:
+            field_names = list(self.columns)
+            if index:
+                field_names = ["Index"] + field_names
+            tuple_type = namedtuple(name, field_names)
+
+        def itertuples_builder(s):
+            if name is not None:
+                values = s._to_pandas().to_dict()
+                if index:
+                    return tuple_type(Index=s.name, **values)
+                return tuple_type(**values)
+            else:
+                values = s._to_pandas().to_numpy()
+                if index:
+                    return tuple([s.name]) + tuple(values)
+                else:
+                    return tuple(values)
```

Comment:
if name is `None` pandas returns named tuple with `name="Pandas"` and fields of `df.columns`. For some reason `np.testing.assert_equal` passes ok with that arguments `np.testing.assert_equal((1,2,3), Pandas(a=1, b=2, c=3))` so the tests did not fails. Maybe another function should be used for testing tuple equality.

Reply:
Default is `name="Pandas` but `name=None` creates a regular tuple: https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.itertuples.html